src/ERC1155A.sol

```diff
@@ -307,37 +276,90 @@ abstract contract ERC1155A is IERC1155A {
         return true;
     }
 
-    /// @notice Internal function for decreasing single id approval amount
-    /// @dev Only to be used by address(this)
-    /// @dev Notice `owner` param, only contract functions should be able to define it
-    /// @dev Re-adapted from ERC20
-    function _decreaseAllowance(
-        address owner,
-        address spender,
+    /*///////////////////////////////////////////////////////////////
+                    SERC20 AND TRANSMUTE LOGIC 
+    //////////////////////////////////////////////////////////////*/
+
+    /// @inheritdoc IERC1155A
+    function registerSERC20(
         uint256 id,
-        uint256 subtractedValue
+        string memory name,
+        string memory symbol
     )
-        internal
+        external
         virtual
-        returns (bool)
+        override
+        returns (address)
     {
-        uint256 currentAllowance = allowance(owner, spender, id);
-        require(currentAllowance >= subtractedValue, "ERC20: decreased allowance below zero");
-        unchecked {
-            _setApprovalForOne(owner, spender, id, currentAllowance - subtractedValue);
+        if (synthethicTokenId[id] != address(0)) revert SYNTHETIC_ERC20_ALREADY_REGISTERED();
+
+        address syntheticToken = address(new sERC20(name, symbol));
```

Comment:
no more decimals so this requires scaling somewhere else

Reply:
added a decimal